TradeDis/frontend
# feat(post): post trade offer

**What was done?**
Story: TRAD-42 "As a user, I should be able to click on a trade post and propose a trade offer to the other trader"

- Clickable posts on Home Feed that leads to Post Screen with more detail on post
- Propose Trade button on Post Screen directed to New Message Screen on press
- Inbox with filler text and options for navigating to New Message and Conversation screens
- Navigation between Home Feed, Post Screen and Inbox screens

**Areas for improvement**

- Message and Conversation routes set up in backend to enable New Message form to work properly
- Possibly remove PostScreen from bottom navigation and combine with HomeFeedScreen into a stack
- Remove usage of hard-coded colours
- Fix styling on PostScreen

**Testing**
Navigation between Home Feed and Post Screen
1. Ensure both the backend server (`yarn dev` in the backend project) and the frontend (`expo start` in the frontend project) are running.
2. On the simulator, go to the Home Feed screen.
3. Click on a post to be directed to Post Screen (Note: Some posts may not show the right information because of missing data in the database.)
4. From the Post Screen, click 'Back' on the navigation bar at the top to return to the Home Feed.

Propose Trade button to New Message Screen
1. Redo steps 1-3 from the test above.
2. Click the "Propose Trade" button to be forwarded to the New Message Screen.
3. Submitting the form should reveal a temporary alert.

Navigation between Inbox and New Message Screen
1. Ensure both the backend server (`yarn dev` in the backend project) and the frontend (`expo start` in the frontend project) are running
2. On the simulator, click on the Inbox tab in the bottom navigation.
3. Once in the Inbox, click on "New" in the menu options. This should lead to the New Message Screen.
4. Click the "Inbox" option on the top navigation bar to return to the Inbox.
5. Clicking on a message in the Inbox should lead to the currently empty Conversation Screen.
6. Repeat step 4 to return back to the Inbox.

Current status of the screens mentioned above.

![localhost_19006_Inbox_Messages](https://user-images.githubusercontent.com/57649385/98517031-50444b80-223b-11eb-8801-ba6c7a4c021a.png)
![localhost_19006_Inbox_Messages (1)](https://user-images.githubusercontent.com/57649385/98517047-576b5980-223b-11eb-9485-0d3a718508e3.png)
![localhost_19006_Inbox_Messages (2)](https://user-images.githubusercontent.com/57649385/98517071-605c2b00-223b-11eb-8cdc-25bba14a256d.png)
![localhost_19006_Inbox_Messages (3)](https://user-images.githubusercontent.com/57649385/98517084-63571b80-223b-11eb-886d-dcbb94e0b979.png)
